### DISCUSSION TITLE
Issues with the Identifier field that is marked as hidden/disabled in UI [e.g OPT-201-Organization-TouchPoint in PIN Info GD]

### DISCUSSION THREAD

**Discussion by VR-M** [OP-TED/eForms-SDK#841] [Q&A] [answered]
Hi,
Checking for PIN Info GD v1.9:
Identifier field like OPT-201-Organization-TouchPoint is not visible in UI or disabled for User interaction in UI has value that is automatically generated.
Use case 1: If the other fields in the Touch-point section are NOT filled, the schematron validation fails if this OPT-201-Organization-TouchPoint is included in the payload.
Use case 1: If the other fields in the Touch-point section are NOT filled, the schematron validation passes if this OPT-201-Organization-TouchPoint is NOT included in the payload.

Are the above used cases correct, if none of the fields in the Touch point section is filled, then the identifier field should not be added to the payload.
And if anyone or all of the fields in this section is/are filled, then this identifier id should be added to the payload.

Or is there any specific condition that i miss?. I look at the metadata [conditions] of this field in fields.json but is not very clear:
{
    "id" : "OPT-201-Organization-TouchPoint",
    "parentNodeId" : "ND-Touchpoint",
    "name" : "TouchPoint Technical Identifier",
    "btId" : "OPT-201",
    "xpathAbsolute" : "/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:Organizations/efac:Organization/efac:TouchPoint/cac:PartyIdentification/cbc:ID",
    "xpathRelative" : "cac:PartyIdentification/cbc:ID",
    "xsdSequenceOrder" : [ { "cac:PartyIdentification" : 3 }, { "cbc:ID" : 2 } ],
    "type" : "id",
    "attributes" : [ "OPT-201-Organization-TouchPoint-Scheme" ],
    "idScheme" : "TPO",
    "schemeName" : "touchpoint",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    "pattern" : {
      "value" : "^TPO-\\d{4}$",
      "severity" : "ERROR"
    },
    "assert" : {
      "value" : "{ND-Root} ${TRUE}",
      "severity" : "ERROR",
      "constraints" : [ {
        "condition" : "{OPT-201-Organization-TouchPoint} ${OPT-201-Organization-TouchPoint is present}",
        "value" : "{OPT-201-Organization-TouchPoint} ${OPT-201-Organization-TouchPoint is unique in /OPT-201-Organization-TouchPoint}",
        "severity" : "ERROR",
        "message" : "rule|text|BR-OPT-00201-0100"
      }, {
        "condition" : "{ND-Organization} ${OPT-201-Organization-TouchPoint is present}",
        "value" : "{ND-Organization} ${((OPT-201-Organization-TouchPoint in OPT-301-Lot-AddInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-DocProvider) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-EmployLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-EnvironLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-FiscalLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-Mediator) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-ReviewInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-ReviewOrg) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-TenderEval) or (OPT-201-Organization-TouchPoint in OPT-301-Lot-TenderReceipt) or (OPT-201-Organization-TouchPoint in OPT-301-Part-AddInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Part-DocProvider) or (OPT-201-Organization-TouchPoint in OPT-301-Part-EmployLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Part-EnvironLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Part-FiscalLegis) or (OPT-201-Organization-TouchPoint in OPT-301-Part-Mediator) or (OPT-201-Organization-TouchPoint in OPT-301-Part-ReviewInfo) or (OPT-201-Organization-TouchPoint in OPT-301-Part-ReviewOrg) or (OPT-201-Organization-TouchPoint in OPT-301-Part-TenderEval) or (OPT-201-Organization-TouchPoint in OPT-301-Part-TenderReceipt) or (OPT-201-Organization-TouchPoint in OPT-301-ReviewBody) or (OPT-201-Organization-TouchPoint in OPT-301-ReviewReq))}",
        "severity" : "ERROR",
        "message" : "rule|text|BR-OPT-00201-0101"
      } ]
    }
  }

**Comment by YvesJo** [comment, marked as answer]
Hi,
Your observations are correct. If the Touchpoint technical identifier is present in the XML while no other Touchpoint information is, then validation will fail. If the Touchpoint technical identifier is not present and no other Touchpoint information is present, then this non-existing touchpoint will not generate any error. This is a similar behavior as with other technical identifiers. 
Your conclusion is correct too: " _if none of the fields in the Touch point section is filled, then the identifier field should not be added to the payload.
And if anyone or all of the fields in this section is/are filled, then this identifier id should be added to the payload._"
When generating the XML also keep in mind that no element may be empty; when it is about a leaf, it must contain some information and when it is a branch, it must contain some elements.
KR

**Reply by VR-M** [reply]
Thanks for your reponse.
